graph-gateway/src/network/subgraph_client/indexers_list_paginated_client/queries.rs

```diff
@@ -0,0 +1,248 @@
+use serde::Deserialize;
+use thegraph_core::types::Attestation;
+use thegraph_graphql_http::{
+    http::{request::IntoRequestParameters, response::Error},
+    http_client::{RequestError, ResponseError, ResponseResult},
+};
+use url::Url;
+
+/// Send an authenticated GraphQL query to a subgraph.
+pub async fn send_query<T>(
+    client: &reqwest::Client,
+    url: Url,
+    auth: &str,
+    query: impl IntoRequestParameters + Send,
+) -> anyhow::Result<ResponseResult<T>>
+where
+    T: for<'de> Deserialize<'de>,
+{
+    let resp = client
+        .post(url)
+        .bearer_auth(auth)
+        .json(&query.into_request_parameters())
+        .send()
+        .await?;
+
+    process_indexer_graphql_response(resp)
+        .await
+        .map_err(anyhow::Error::from)
+}
+
+#[derive(Debug, Deserialize)]
+pub struct IndexerResponsePayload<T> {
+    #[serde(rename = "graphQLResponse")]
+    pub data: Option<T>,
```

Comment:
This should be `data: Option<String>` which then gets parsed into T after receiving it.

Reply:
Isn't this what _serde_ does? 

- If the field is not present, it sets the field to `None`.
- If it is present, it deserializes it into the `T` type.

Reply:
Ok, I just saw the error message. 

It seems that we are escaping the JSON object as a string in the indexer service... Another deviation from the standard 😓